File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/csharp/scc.cs
/**
 * C# Implementation: Strongly Connected Components (Tarjan & Kosaraju)
 * Invariant: O(V + E) single or double DFS passes
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class StronglyConnectedComponentsTarjanKosaraju<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/cpp/scc.cpp
/**
 * C++ Implementation: Strongly Connected Components (Tarjan & Kosaraju)
 * Invariant: O(V + E) single or double DFS passes
 */
#include <iostream>
#include <vector>
#include <stdexcept>

template <typename T>
class StronglyConnectedComponentsTarjanKosaraju {
private:
    std::vector<T> elements;
public:
    // TODO: Implement primary invariant operations
    void push(const T& val) {
        elements.push_back(val);
    }
    size_t size() const { return elements.size(); }
};
